ui/pages/swaps/transaction-settings/transaction-settings.js

```diff
@@ -0,0 +1,332 @@
+import React, { useState, useEffect, useContext } from 'react';
```

Comment:
Most of the code here is reused from the SlippageButtons component: https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/swaps/slippage-buttons/slippage-buttons.js